### PR TITLE
flushing when payload max size has reached

### LINKED ISSUE #276: Flush when maximum storage size limit has been reached

### PR DESCRIPTION
**Pull request by pbassut**
Closes #276 #219

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ class Analytics {
     this.axiosInstance = axiosInstance
     this.timeout = options.timeout || false
     this.flushAt = Math.max(options.flushAt, 1) || 20
+    this.maxQueueSize = options.maxQueueSize || 1024 * 32 // defaults to 32kb
```
Comment (pooyaj) [inline index.js:48]: shall we use larger queue size based on the feedback we got on the other libraries?

Comment (pbassut) [reply index.js:48]: We should. I'll make the change

**Review by pooyaj — COMMENTED** [review]
@pbassut looks like one of the tests are failing 🤔 can you please check that before we merge it?

**Comment by pbassut** [comment]
I had to remove support for node 8 since one of our libraries dropped support for it already.
If this is a concern, maybe we should rethink how we're going to do http requests with another library.

**Review by pooyaj — APPROVED** [review]
Let's drop Node 8 support as it is already EOL-ed for a while.

**Pull request merged**